server/catgenome/src/main/java/com/epam/catgenome/manager/externaldb/PdbDataManager.java

```diff
@@ -46,82 +56,140 @@
 
-        ParameterNameValue[] params = new ParameterNameValue[]{new ParameterNameValue("query", pdbIds)};
+            return parseToDasalignment(QueryUtils.execute(rcsbApi.getDasalignment(query)));
 
-        String location = RCSB_SERVER + RCSB_PDP_MAP_TOOL + "?";
+        } catch (RSCBResponseException e) {
+            throw new ExternalDbUnavailableException(MessageHelper.getMessage(MessagesConstants
+                    .ERROR_UNEXPECTED_FORMAT, LOCATION), e);
+        }
+    }
 
-        String rcsbpbdData = httpDataManager.fetchData(location, params);
+    private String replaceHttpSymbols(String query, String pdbIds) {
+        return String.format(query, pdbIds)
+                .replaceAll("\\{", "%7B")
+                .replaceAll("}", "%7D")
+                .replaceAll(" ", "%20");
+    }
 
-        try {
-            JAXBContext jaxbContext = JAXBContext
-                    .newInstance("com.epam.catgenome.manager.externaldb.bindings.ecsbpdbmap");
-            Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+    private Dasalignment parseToDasalignment(DasalignmentDTO das) {
+        List<Alignment> alignments = new ArrayList<>();
+
+        List<PolymerEntity> polymerEntities = das.getData().getEntry().getEntities();
+        for (PolymerEntity entity : polymerEntities) {
+            for (PolymerEntityInstance instance : entity.getEntityInstances()) {
+                String id = instance.getId();
+                List<PdbBlock> pdbBlocks = new ArrayList<>();
+
+                for (InstanceFeature feature : instance.getInstanceFeatures()) {
+                    List<Segment> segments = new ArrayList<>();
+
+                    for (FeaturePosition position : feature.getPositions()) {
+                        Integer start = position.getStart();
+                        Integer end = position.getEnd();
+                        segments.add(parseSegment(start, end, id));
+                    }
+                    pdbBlocks.add(parsePdbBlock(segments));
+                }
+                alignments.add(parseAlignment(pdbBlocks));
+            }
+        }
 
-            StringReader reader = new StringReader(rcsbpbdData);
-            Object pdbObject = unmarshaller.unmarshal(reader);
+        Dasalignment dasalignment = new Dasalignment();
+        dasalignment.setAlignment(alignments);
+        return dasalignment;
+    }
 
-            Dasalignment fetchedDasalignmentData = null;
-            if (pdbObject instanceof Dasalignment) {
-                fetchedDasalignmentData = (Dasalignment) pdbObject;
+    private Dataset parseToDataset(DatasetDTO datasetDTO, String structureId) {
+        List<Record> records = new ArrayList<>();
+
+        Entry entry = datasetDTO.getData().getEntry();
+        String structureTitle = entry.getStruct().getTitle();
+        String classification = entry.getKeywords().getValue();
+
+        for (PolymerEntity entity : entry.getEntities()) {
```

Comment:
try to rewrite it with `Stream.map` and `Stream.flatMap`

Reply:
Fixed, as possible!